### DISCUSSION TITLE
timelineItems doesn't work (no results)

### DISCUSSION THREAD

**Discussion by jesuslinares** [community/community#57326] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

I am encountering some issues when trying to retrieve the event history of the issues added to projects (v2). Example:

> [user] moved this from In progress to Pending review in [project] [3 weeks ago]

Specifically, I am using the timelineItems objects, but it seems that only "labeled" events are being displayed. I have already enabled the project-event-details-preview as mentioned in the documentation (https://docs.github.com/en/graphql/overview/schema-previews#project-event-details-preview). Also, I checked that I have enough permissions.

However, I am unable to access the desired event history. Could you please assist me in understanding how to retrieve the complete event history for the issues in question?

This is the query:

```
query myquery($project_id: ID!, $cursor: String) {
  node(id: $project_id) {
    ... on ProjectV2 {
      items(first: 100, after: $cursor) {
        ...
        nodes {
          ...
          content {
            ... on Issue {
              ...
              timelineItems(first: 100) {
                nodes {
                  __typename
                  ... on AddedToProjectEvent {
                    actor { login }
                    projectColumnName
                  }
                  ... on MovedColumnsInProjectEvent {
                    actor { login }
                    createdAt
                    project { name }
                    databaseId
                    previousProjectColumnName
                    projectColumnName
                  }
                  ... on LabeledEvent {
                    actor { login }
                    label { name } 
                  }
                  ... on UnlabeledEvent {
                    actor { login }
                    label { name } 
                  }
                }
              }
...
```

Thank you for your attention to this matter.

**Comment by ghostinhershell** [comment]
Hey @jesuslinares, thanks so much for reaching out and welcome to the community! I've shared this with our internal teams for feedback and should have feedback for you soon. Thanks again!

**Reply by jesuslinares** [reply]
Thank you for looking into it, any progress on this? Thanks!

**Reply by mifrazmurthaja** [reply]
Hello @ghostinhershell, do we have an ETA for when this issue will be resolved? The current situation is impeding the implementation of https://github.com/orgs/community/discussions/49602#discussioncomment-5849357

Thanks!

**Reply by jesuslinares** [reply]
We also have similar implementations that are blocked due to this issue.

**Reply by cuongleqq** [reply]
Hello @ghostinhershell, it's been several months. Any progress on fixing the issue?

**Reply by r4fx** [reply]
Hey @ghostinhershell, could you share any update?

**Reply by hianhianhian** [reply]
Could we get an update on this issue? Our team is depending on this API but are currently blocked by this.

**Reply by gmjuhasz** [reply]
+1 on this, experiencing the same issue.

**Reply by rmh54** [reply]
Same here - still an issue

**Reply by qkdreyer** [reply]
@ghostinhershell any updates ?

**Reply by critch** [reply]
I found an odd situation. 

We converted from a classic project board to a v2 board. Issues that had been on a classic board show events like moved. But, then any issues created on only the V2 board do not show these events.

So if I try to mirror the moves on the now close classic board still show up with a projectV2 graphql query.

**Reply by mlisivka** [reply]
@ghostinhershell Any updates on this? This is critical for me

**Reply by srbarrios** [reply]
Same issue here, it seems to only show this value in the cards migrated from the classic project.

**Reply by lilachn** [reply]
@ghostinhershell are there any updates regarding this API issue? Even an answer like - "no, it won't be fixed soon" will be very helpful to know where we stand. I saw that there are more such open issues, this is not the only thread. Thanks

**Reply by srbarrios** [reply]
> @ghostinhershell are there any updates regarding this API issue? Even an answer like - "no, it won't be fixed soon"

That's the e-mail I received from them:

_GitHub (GitHub Support)

Nov 21, 2024, 11:58 AM UTC

Hi Oscar,
 
Thank you for reaching out!
 
Yeah, that behaviour is currently expected. The MOVED_COLUMNS_IN_PROJECT events is specific to legacy project which is now deprecated. Similar events has not yet been implemented in ProjectV2. This is on the team's radar, and will be implemented in the future, but at this time we do not have an ETA. I have added your report as a +1 on the internal issue tracking this as a feature request.
 
I recommend you keep an eye on the changelog and blog where we make announcements about new features and fixes.
 
Please let me know if there is anything else I can assist with.
 
Regards,
 
Oluwaseun
GitHub Support_

**Reply by Coroxx** [reply]
Any updates ?

**Reply by daniyalfarman** [reply]
Any update?

**Comment by naefp** [comment]
I came here because I'm trying to migrate our automation from classic projects to v2 (and it looks like I have 1 month left to do it...)

we are using timeline events as well and I'm trying to reverse engineer what Github UI does:
- github performs an internal query and receives the events on the property `frontTimeline` which is not available via the public API
- the totalcount on frontTimeline equals to totalcount on `timelineItems` (13 in my case), but Github UI does not load timelineItems at all
- i can find these events with type `ProjectV2ItemStatusChangedEvent`, which i cannot find in any documentation, qraphql explorer or somewhere else than in the github internal results
- when i get an issue via the api with timelineItems my totalcount is lower than expected (6 in my case) and the status changed events are missing

For me it's unfortunate that I can't get all the information via the API for V2 projects, but the classic projects go down in a month... ?

**Reply by naefp** [reply]
btw. it seems like the github iOS app is affected by this as well. cant see any projectv2 events in the issue timeline, but I only see label or milestone events..

**Reply by ericboehs** [reply]
I imagine once the iOS app starts listing these events, a new project type will be dropped in the [GraphQL schema changelog](https://docs.github.com/en/graphql/overview/changelog).

I'm wanting to create a sprint report and knowing when an issue was added to a Sprint would be great.

**Comment by Coroxx** [comment]
Hey, idk if it cans help we use a polling everyday at midnight to add a label on the concerned Issue. (For example, issue no. 242 was added today to the “Backlog” project. We therefore add a label named “Backlog” and delete it if it is no longer present in the project).

It allows us to get our timelines with Unlabeled and Labeled events 👍

**Comment by andreaslech** [comment]
I saw your note about using a daily midnight poll to add/remove labels for tracking when issues enter or leave a project column — that’s a clever way to leverage Labeled/Unlabeled events for timelines 👍 

We had a similar need, so I set up periodic polling (shorter intervals) to capture and store actual board status transitions over time. That lets us calculate metrics like **Cycle Time** and **Time in State**, which our Scrum Masters use regularly. 

I wrapped this into a small web app that visualizes issue movements. Overview here: [Flow2C.com](https://flow2c.com) — or jump straight into the demo: [app.flow2c.com](https://app.flow2c.com). 


In your workflows, would historical analytics alone be enough, or would near‑real‑time notifications on status changes also be useful?

**Comment by rileybroughten** [comment]
You'll now see additional events in GraphQL including `Status` changes on an issue. Check out the [changelog](https://github.blog/changelog/2025-11-06-improved-onboarding-flow-for-github-projects/#%f0%9f%a4%96-api-improvements) for more details!

**Comment by ericboehs** [comment]
This is excellent news! I had Claude run some queries via `gh api` on my test board. Here's what it found:

<details><summary>Summary: GitHub Projects V2 API - What's Available NOW</summary>

### ✅ Timeline Events (FULLY WORKING)

1. **`AddedToProjectV2Event`**
   - Fields: `id`, `createdAt`, `actor`, `project`, `wasAutomated`
   - Tracks when items are added to a project
   - Distinguishes automated vs manual additions

2. **`ProjectV2ItemStatusChangedEvent`**
   - Fields: `id`, `createdAt`, `actor`, `previousStatus`, `status`, `wasAutomated`, `project`
   - Tracks every status transition with before/after values
   - **THIS IS THE KEY EVENT** for your sprint reporting

3. **`RemovedFromProjectV2Event`**
   - Fields: `id`, `createdAt`, `actor`, `project`, `wasAutomated`
   - Tracks when items are removed from a project

4. **`ConvertedFromDraftEvent`**
   - Fields: `id`, `createdAt`, `actor`
   - Tracks draft → issue conversions

### ✅ Query Filtering (WORKING)

**Supported query patterns:**
- `status:Ready` - Filter by single status
- `status:"In Progress"` - Status with spaces (must quote)
- `status:Ready,"In Progress"` - OR logic (9 items in your project)
- `status:Ready priority:P0` - AND logic (combine filters)
- `priority:P0` - Filter by custom fields

**Does NOT work (or unknown syntax):**
- Date range filtering like `updated:>2025-11-06`
- Assignee filtering may have different syntax

## Your Sprint Reporting Possibilities

### 1. Track Mid-Sprint Item Movement
```graphql
# Query all status changes for an issue
issue(number: 38) {
  timelineItems(itemTypes: [PROJECT_V2_ITEM_STATUS_CHANGED_EVENT]) {
    nodes {
      ... on ProjectV2ItemStatusChangedEvent {
        createdAt
        actor { login }
        previousStatus
        status
        wasAutomated
      }
    }
  }
}
```

**Result:** You can see exactly when someone moved issue #38 from `"Backlog"` → `"Upcoming Sprint"` → `"Ready"`

### 2. Calculate Time-in-Status

From the actual data for issue #34:
- **In Progress:** Nov 4, 21:03 → Nov 5, 22:55 = **1.08 days**
- **Total lead time:** Oct 30, 17:16 → Nov 5, 22:55 = **6.23 days**

### 3. Filter Project by Status

```graphql
projectV2(number: 20) {
  currentSprint: items(query: "status:\"Current Sprint\"") {
    totalCount  # How many items in current sprint
  }
  done: items(query: "status:Done") {
    totalCount  # How many completed (27 in your project)
  }
}
```

### 4. Track Who Made Changes

Every event has `actor.login`, so you can track:
- Who moves items to `"Done"`
- Who adds items mid-sprint
- Who moves items backwards (rework)

### 5. Detect Automation vs Manual

Every event has `wasAutomated` boolean:
- Manual changes: `wasAutomated: false`
- Automated: `wasAutomated: true` (like `"github-project-automation"`)

## Practical Example for Your Use Case

**Query to build sprint report:**

```graphql
{
  organization(login: "software") {
    projectV2(number: 20) {
      # Get all items that were in current sprint
      items(first: 100, query: "status:\"Current Sprint\",Done") {
        nodes {
          content {
            ... on Issue {
              number
              title
              # Get the complete timeline
              timelineItems(first: 100, itemTypes: [
                ADDED_TO_PROJECT_V2_EVENT,
                PROJECT_V2_ITEM_STATUS_CHANGED_EVENT
              ]) {
                nodes {
                  ... on AddedToProjectV2Event {
                    createdAt
                    wasAutomated
                  }
                  ... on ProjectV2ItemStatusChangedEvent {
                    createdAt
                    actor { login }
                    previousStatus
                    status
                    wasAutomated
                  }
                }
              }
            }
          }
        }
      }
    }
  }
}
```

**This would give you:**
1. All items currently or previously in the sprint
2. Complete history of when each entered the project
3. Every status transition with timestamps
4. Who made each change
5. Which changes were automated
</details>

Thanks for getting this in the API! Going to make Sprint Reporting much easier for my team. 🙂

**Comment by ericboehs** [comment]
It looks like it's just for THE `Status` field and not other fields like Sprint (Iteration), Estimate (Number), or SingleSelects that we use for Column By. Is that correct?

It would be great to query when an issue is moved into/out of a Sprint. I think right now we have to poll or use a webhook for that.

**Reply by rileybroughten** [reply]
@ericboehs thank you for the feedback, and correct this just includes the `Status` field, not any other project custom fields at this point
